flocker/node/agents/_logging.py

```diff
@@ -94,3 +99,70 @@
     [VOLUME_ID, STATUS, TARGET_STATUS, WAIT_TIME],
     u"Waiting for a volume to reach target status.",)
 # End: Helper datastructures used by AWS storage driver.
+
+# Begin: Helper datastructures used by OpenStack storage drivers
+
+CODE = Field.for_types("code", [int], u"The HTTP response code.")
+MESSAGE = Field.for_types(
+    "message", [bytes, unicode],
+    u"A human-readable error message given by the response.",
+)
+DETAILS = Field.for_types("details", [dict], u"Extra details about the error.")
+REQUEST_ID = Field.for_types(
+    "request_id", [bytes, unicode],
+    u"The unique identifier assigned by the server for this request.",
```

Comment:
Does this need to support both types?  There should usually be a specific (good) reason for allowing bytes and unicode to mix - and it should be explained in the code that allows the mixing.  And then all variations of mixing need to be tested.  Python makes it very easy to shoot yourself in the foot in this area by providing implicit conversion between **some** byte strings to unicode and **some** unicode strings to bytes.  This implicit conversion makes it easy to conclude that your string usage is correct only to learn later on that it's totally broken (because someone decided to run your code in a French locale or some such).